### PR TITLE
THRIFT-5519 Java async client loses exceptions in void methods

### PR DESCRIPTION
**Pull request by alex-kormukhin**
Fix [THRIFT-5519 Java async client loses exceptions in void methods](https://issues.apache.org/jira/browse/THRIFT-5519).

### CODE REVIEW

File: test/VoidMethExceptionsTest.thrift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+exception TExampleException {
+  1: required string message;
+}
+
+service TAppService01 {
+  string returnString(1: string msg, 2: bool throwException) throws (1:TExampleException error);
+  void returnVoidThrows(1: string msg, 2: bool throwException) throws (1:TExampleException error);
+  void returnVoidNoThrowsRuntimeException(1: string msg, 2: bool throwException);
+  void returnVoidNoThrowsTApplicationException(1: string msg, 2: bool throwException);
+  oneway void onewayVoidNoThrows(1: string msg, 2: bool throwException);
```
Comment (Jens-G) [inline test/VoidMethExceptionsTest.thrift:12]: Just to be clear: A oneway does not return anything. Including exceptions. There is no return. None.

Comment (alex-kormukhin) [reply test/VoidMethExceptionsTest.thrift:12]: Yes, test checks exactly that (no client-side exceptions when server throws, no client-side wait when server hangs). Method name reflects correct signature for such methods.

**Comment by alex-kormukhin** [comment]
Try to fix this with https://github.com/apache/thrift/pull/2536, but may be infrastructure problem (https://github.com/johnrengelman/shadow/issues/221).

**Comment by Jens-G** [comment]
Could you force-push? That will trigger another CI run.

**Comment by Jens-G** [comment]
Tons of "no @param for ..." warnings

**Pull request merged**